service/resource/configmap/update.go

```diff
@@ -40,20 +40,24 @@ func (r *Resource) GetUpdateState(ctx context.Context, obj, currentState, desire
 	// If the current state does not match the desired state,
 	// return the desired state as update.
 	if currentConfigMap.Data[r.configMapKey] != desiredConfigMap.Data[r.configMapKey] {
-		return nil, desiredConfigMap, nil, nil
+		r.logger.Log("debug", "current configmap does not match desired configmap, need to update")
+		return nil, nil, desiredConfigMap, nil
```

Comment:
This is the error. `operatorkit` expects create, delete, update, i was going with create, update, delete. All good.